src/app/beer_garden/events/processors.py

```diff
@@ -125,16 +128,47 @@ def register(self, processor, manage: bool = True):
 class HttpEventProcessor(QueueListener):
     """Publish events using an EasyClient"""
 
-    def __init__(self, easy_client=None, black_list=None, **kwargs):
+    def __init__(
+        self, easy_client=None, black_list=None, reconnect_action=None, **kwargs
+    ):
         super().__init__(**kwargs)
 
         self._ez_client = easy_client
         self._black_list = black_list or []
+        self._reconnect_action = reconnect_action
 
     def process(self, event: Event):
+
         try:
             if event.name not in self._black_list:
                 event.garden = beer_garden.config.get("garden.name")
                 self._ez_client.publish_event(event)
         except Exception as ex:
             logger.exception(f"Error publishing EasyClient event: {ex}")
+
+            self.reconnect()
+
+    def reconnect(self):
+
+        # Mark not processing and stop accepting events
+        self._processing = False
+
+        # Purge the current Queue
+        while not self._queue.empty():
+            self._queue.get()
+
+        # Back-off connection from EZ Client
+        wait_time = 0.1
+        while not self.stopped() and not self._processing:
+            if self._ez_client.can_connect():
+                self._processing = True
+
+                if self._reconnect_action:
+                    self._reconnect_action()
+
+            else:
+                self.logger.warning(
+                    "Waiting %.1f seconds before next attempt", wait_time
+                )
+                self.wait(wait_time)
+                wait_time = min(wait_time * 2, 30)
```

Comment:
Hmm, we should probably make _progressive_backoff a member of `StoppableThread` instead of the application. We'd have to tweak it to allow setting the log level of the output messages, though. Maybe not worth the effort.

Reply:
I am not going to do that for this PR.